xmtp/src/lib.rs

```diff
@@ -28,12 +30,25 @@ pub trait Errorer {
 }
 
 pub trait Store<I> {
-    fn store(&self, into: &mut I) -> Result<(), String>;
+    fn store(&self, into: &mut I) -> Result<(), StorageError>;
+}
+
+pub trait KeyStore: Default + Errorer {
+    fn get_account(&mut self) -> Result<Option<Account>, StorageError>;
+
+    fn set_account(&mut self, account: &Account) -> Result<(), StorageError>;
+}
+
+#[derive(Debug, Error)]
+pub enum StorageError {
+    #[error("could not Fetch: {0}")]
+    Fetch(String),
+    #[error("could not Store: {0}")]
+    Store(String),
```

Comment:
Rather than using an associated error type, the trait impls all return a `StorageError` for easier type management

Reply:
Great minds think alike. I did almost the exact same thing in my branch.